packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -687,7 +688,16 @@ class _ViewContentState extends State<_ViewContent> {
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
-    result = widget.suggestionsBuilder(context, _controller);
+    final FutureOr<Iterable<Widget>> suggestionsFutureOr = widget.suggestionsBuilder(context, _controller);
```

Comment:
Can you make this `async` and then do this?

```
result = await widget.suggestionsBuilder(context, _controller);
```

Reply:
Is it right? `didChangeDependencies()` overrides a sync function.

Reply:
I think it works?  The code below compiles for me.  Or any reason not to?

```dart
class Foo extends StatefulWidget {
  const Foo({
    super.key,
  });

  @override
  State<Foo> createState() => _FooState();
}

class _FooState extends State<Foo> {
  @override
  void didUpdateWidget(Foo oldWidget) async {
    super.didUpdateWidget(oldWidget);

    await Future.delayed(const Duration(seconds: 1));
    print('delayed hello');
  }

  @override
  Widget build(BuildContext context) {
    return Container(
      child: const Text('Foo'),
    );
  }
}
```

Reply:
Should we also change the function type to `Future<void> didChangeDependencies`? I tried it locally and the linter shows a warning about its type. This works for me😀:
<img width="851" alt="Screenshot 2023-05-17 at 3 50 22 PM" src="https://github.com/flutter/flutter/assets/36861262/0203037a-07a7-48cc-88b1-cd21bb3bf220">

Reply:
Oh, I didn't realize that it could be placed on the last line to avoid using context across async.

I'm trying `if (!mounted) {return;}`.